iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -438,30 +441,19 @@ static EXPECTED_SEND_DATA* EventData_Create(void)
 static EXPECTED_SEND_DATA* EventData_Create_With_Custom_Size(size_t messageSize)
 {
     EXPECTED_SEND_DATA* result;
-    char* message = malloc(sizeof(char) * messageSize);
+    char* message = calloc(messageSize, sizeof(char));
 
-    if (message == NULL)
-    {
-        LogError("Failed to allocate EventData message");
-        result = NULL;
-    }
-    else
-    {
-        char idString[10];
-        size_t idStringLength = sprintf(idString, "%d", (int)g_iotHubTestId++);
+    ASSERT_IS_NOT_NULL(message, "Failed to allocate EventData message");
 
-        (void)memset(message, (int)'a', messageSize - 1);
-        (void)memcpy(message + 1, idString, idStringLength);
-        message[messageSize - 1] = '\0';
+    char idString[10];
```

Comment:
Nit - couldn't you `sprintf` into the `message` buffer directly instead of needing a separate `idString`?

Reply:
Taken.